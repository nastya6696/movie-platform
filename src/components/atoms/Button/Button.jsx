import styles from './styles.module.scss';

export const Button = ({name}) => (
  <button className={styles.btnBasic}>{name}</button>
)
