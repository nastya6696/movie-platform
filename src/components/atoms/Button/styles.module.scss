@import "src/assets/scss/common";

.btnBasic {
  width: 233px;
  height: 48px;
  background-color: $red;
  color: $white;
  border: none;
  border-radius: 4px;
  @include fontSizes(20px, 25px);
  font-weight: 200;
}
