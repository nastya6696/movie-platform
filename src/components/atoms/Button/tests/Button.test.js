import { create } from 'react-test-renderer';
import {Button} from "../Button";

describe('Button', () => {
  it('should create simple button component', () => {
    const handleClick = jest.fn();
    const component = create(<Button name='Test button name' handleClick={handleClick} />);

    expect(component).toMatchSnapshot();
  })
})
