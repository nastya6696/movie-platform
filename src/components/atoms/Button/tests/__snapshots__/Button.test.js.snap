// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should create simple button component 1`] = `
<button
  className="btnBasic"
  onClick={[MockFunction]}
>
  Test button name
</button>
`;
